sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2197,13 +2197,23 @@ object functions {
   def round(e: Column): Column = round(e, 0)
 
   /**
-   * Round the value of `e` to `scale` decimal places with HALF_UP round mode
-   * if `scale` is greater than or equal to 0 or at integral part when `scale` is less than 0.
+   * Returns the value of the column `e` rounded to 0 decimal places with HALF_UP round mode.
    *
    * @group math_funcs
    * @since 1.5.0
    */
-  def round(e: Column, scale: Int): Column = withExpr { Round(e.expr, Literal(scale)) }
+  def round(e: Column, scale: Int): Column = round(e, scale, "half_up")
+
+  /**
+   * Round the value of `e` to `scale` decimal places with given round mode, default: HALF_UP
+   * if `scale` is greater than or equal to 0 or at integral part when `scale` is less than 0.
+   *
+   * @group math_funcs
+   * @since 3.3.0
+   */
+  def round(e: Column, scale: Int, mode: String): Column = withExpr {
```

Comment:
Is it better for the types of all the arguments to be `Column` for newly added functions?
Please see the comment at the head of `functions.scala`.

Reply:
I see this function mostly as a companion to the previously existing function than a completely new function. Otherwise it would be a bit weird the round function with 2 arguments takes scale parameter in `Int` type but the round function with 3 arguments takes the scale parameter in `Column` type. What do you think ?

Reply:
I think, it's better to make the arguments be `Column` for the overloaded one.
What do you think @HyukjinKwon ?

Reply:
ok, i'll do the modifications.